metadata-ingestion/src/datahub/ingestion/source/sql/snowflake.py

```diff
@@ -748,6 +754,38 @@ def _is_dataset_allowed(
             return False
         return True
 
+    def generate_profile_candidates(
+        self, inspector: Inspector, threshold_time: datetime
+    ) -> List[str]:
+
+        _profile_candidates = []
+
+        db_rows = inspector.engine.execute(
+            text(
+                """
+SELECT TABLE_CATALOG, TABLE_SCHEMA, TABLE_NAME
+FROM INFORMATION_SCHEMA.TABLES
```

Comment:
Does access to this table require any specific privilege? If yes, please ensure that is added in snowflake docs and `provision_role` block in snowflake is updated.

Reply:
According to my understanding, this does not require any additional priviledges. The same table is already being accessed for fetching details using snowflake sqlalchemy dialect.